python/lsst/ts/observatory/control/auxtel/atcalsys.py

```diff
@@ -208,6 +212,144 @@ async def prepare_for_flat(self, config_name: str) -> None:
             task_setup_electrometer,
         )
 
+    async def calculate_optimized_exposure_times(
+        self, wavelengths: list, config_data: dict
+    ) -> list[ATCalsysExposure]:
+        """Calculates the exposure times for the fiber spectrograph and
+        the electrometer given camera exposure times and other
+        relevant configuration paramaters.
+
+        This includes wavelength dependent information for the fiber
+        spectrograph and also includes the logic for the size of
+        the electrometer buffer.
+
+        Parameters
+        ----------
+        wavelengths : `list`
+            List of all wavelengths for this exposure list
+        config_data : `dict`
+            All information from configuration file
+
+        Returns
+        -------
+        exposure_list : `array`
+            List of exposure information, including wavelength and camera,
+            fiberspectrograph and electrometer exposure times.
+
+        """
+
+        exposures = []
```

Comment:
You might want to provide the type annotation here. Mypy is probably able to derive it from context, but it doesn't hurt to help it out. It also let whoever is reading the code know that this variable has the same type as the return type, so it is probably the variable you are returning.